compiler/src/dmd/nogc.d

```diff
@@ -103,8 +103,23 @@ public:
             return;
         if (f.setGC())
         {
-            e.error("array literal in `@nogc` %s `%s` may cause a GC allocation",
-                f.kind(), f.toPrettyChars());
```

Comment:
Why not just print the expression here and be done with it? Adding all this extra complexity in the compiler for the promise of a benefit doesn't make sense in my opinion.

Reply:
Because the error message is unreadable unless you are lucky enough that the compiler forgets it's done the semantic analysis